src/snapshot/checksum.rs

```diff
@@ -0,0 +1,61 @@
+use std::{
+    collections::hash_map::DefaultHasher,
+    hash::{Hash, Hasher},
+    marker::PhantomData,
+};
+
+use bevy::prelude::*;
+
+use crate::{SaveWorld, SaveWorldSet};
+
+/// Flags an entity as containing a checksum for a type `T`
+#[derive(Component)]
+pub struct ChecksumFlag<T> {
+    _phantom: PhantomData<T>,
+}
+
+impl<T> Default for ChecksumFlag<T> {
+    fn default() -> Self {
+        Self {
+            _phantom: default(),
+        }
+    }
+}
+
+/// Represents a checksum value for a specific type, flagged by [`ChecksumFlag`].
+#[derive(Component, Default, Hash)]
+pub struct ChecksumPart(pub u64);
+
+/// Represents a total checksum for a given frame.
+#[derive(Resource, Default, Clone, Copy)]
+pub struct Checksum(pub u64);
```

Comment:
`u128` is used by ggrs, so maybe use that here as well?

Reply:
Makes sense, I was following the type returned by `Hash`, but we don't actually know that `Hash` is even a good fit for the checksum anyway.